### PR TITLE
1870: Add the PriceProtection step to the OR

### PR DESCRIPTION
**Pull request by daniel-sousa-me**
When there's an ebuy, if shares are sold, the president can also protect
them. Had to add the step to the OR and that required changing some
assumptions.

### CODE REVIEW

File: lib/engine/game/g_1870/step/price_protection.rb [hunk 1]
```diff
@@ -22,7 +22,11 @@ def description
           end
 
           def round_state
-            { sell_queue: [] }
+            super.merge(sell_queue: [])
+          end
+
+          def active_entities
+            @round.sell_queue.map(&:president)
```
Comment (jenf) [inline lib/engine/game/g_1870/step/price_protection.rb:29]: This should only return one active entity.

File: lib/engine/game/g_1870/step/price_protection.rb [hunk 2]
```diff
@@ -60,24 +66,23 @@ def process_buy_shares(action)
               price: price
             )
 
-            @round.goto_entity!(player)
+            @round.goto_entity!(player) if @round.entities[@round.entity_index].player?
```
Comment (jenf) [inline lib/engine/game/g_1870/step/price_protection.rb:69]: Why this change? Or is it to stop the entity changing in an OR?

Comment (daniel-sousa-me) [reply lib/engine/game/g_1870/step/price_protection.rb:69]: If player B sells a share of a corp of player A, and player A protects the share, the game continues to the player next to player A (instead of the one next to player B!)

But this only makes sense in the SR

Comment (jenf) [reply lib/engine/game/g_1870/step/price_protection.rb:69]: Please add a comment explaining.

**Review by jenf — APPROVED** [review]
